Add support for installing a `deb` instead of `tar.gz`

philwills opened the pull request.

Commit:
Add support for installing a `deb` instead of `tar.gz`

philmcmahon commented:
Awesome! Does the dpkg command always start the app?

philwills commented:
That depends on how you set up the `deb`. The one's produced by sbt-native-packager, which are what I've used do create a systemd file which will start on installation.

philmcmahon commented:
Sounds good. 👍

philwills commented on packer/resources/features/deb/install.sh:
```diff
@@ -0,0 +1,73 @@
+# Install an application that was packaged by the sbt-native-packager
+# download
+LOCAL_DEB=$(mktemp --suffix=".${TYPE}" /tmp/app-deb.XXXXXX)
+aws s3 cp "s3://${BUCKET}/${STACK}/${STAGE}/${APP}/${APP}_1.0_all.deb" \
```
The `_1.0_all` bit is ugly. I'm asking on the sbt-native-packager gitter if there's a sensible way to customise this, as it annoys me.

philwills commented:
Diff for when I tested it on status app: https://github.com/guardian/ophan/compare/machine-images-deb?expand=1

sihil commented:
Does the 1.0 come from the version of the app?

On Wed, 31 Aug 2016 17:26 Phil Wills, notifications@github.com wrote:

> Diff for when I tested it on status app:
> https://github.com/guardian/ophan/compare/machine-images-deb?expand=1
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/guardian/machine-images/pull/117#issuecomment-243820152,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ABLd8q8PS4gU4-rEDSh5Fk0bQVRWWrpUks5qlasagaJpZM4Jxy4y
> .

## 

---

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.

Guardian News & Media Limited is a member of Guardian Media Group plc. Registered 
Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP.  Registered 
in England Number 908396

philwills commented:
Yea, it comes from he version, as specified in the build.sbt. I'm happen to go with another arbitrary default value.

sihil commented:
Feels like it should be specifiable in the script as not everyone will choose `1.0`. For now can you not rename it by overriding `riffRaffArtifactResources` in the `sbt-riffraff-artifact` config?

philwills commented:
Good points both. I'll see what I can do.

Commit:
Don't do anything magic with deb names

philwills commented:
@sihil I've updated it so the existing `-a` flag is actually used, so can be used for overriding the deb name and removed any auto-suffixing. 

I'll try and actually build an AMI to test that with next week.

sihil commented:
Nice. I think we should make a concerted effort to move from machine-images to AMIgo at some stage.

philwills commented:
Have confirmed this works.
